### PR TITLE
Mp nonuni tiling

### PR DESCRIPTION
**Pull request by anithapk**
Hi Jenny, this PR includes changes for accomplishing the following tasks:
1) include mask generation and tiling support for datasets with non-uniform structure .i.e, variable number of time points and slices per position (assumes same structure across channels) and multi-processing support for tiling + tests
2) Add multi-processing support for mask generation 
3) Split the input module into preprocess and input modules.
I'm really sorry about the length of this PR 😺 !

### CODE REVIEW

**Comment by smguo** [comment]
`        if 'num_workers' in pp_config['tile']:
            num_workers = pp_config['num_workers']`

Is this supposed to be 

`        if 'num_workers' in pp_config['tile']:
            num_workers = pp_config['tile']['num_workers']`

File: micro_dl/utils/aux_utils.py [hunk 1]
```diff
@@ -134,7 +135,7 @@ def sort_meta_by_channel(frames_metadata):
         channel X.
     """
 
-    metadata_ids = validate_metadata_indices(
+    metadata_ids, tp_dict = validate_metadata_indices(
```
Comment (smguo) [inline micro_dl/utils/aux_utils.py]: changed this to match the function output

File: micro_dl/utils/tile_utils.py [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+               tile_size,
+               step_size,
+               isotropic=False,
+               return_index=False,
+               min_fraction=None,
+               save_dict=None):
+    """
+     Tiles the image based on given tile and step size.
+
+     USE MIN_FRACTION WITH INPUT_IMAGE.DTYPE=bool
```
[the remark this answers is not in the document]
Comment (anithapk) [reply micro_dl/utils/tile_utils.py]: Yes :-)

File: micro_dl/preprocessing/tile_images_uni_struct.py [hunk 3]
```diff
@@ -0,0 +1,591 @@
...
+
+            # tile_image uses min_fraction assuming input_image is a bool
+            mask_meta_df_list = mp_tile_save(mask_fn_args,
+                                             workers=self.num_workers)
+            mask_meta_df = pd.concat(mask_meta_df_list, ignore_index=True)
+            # Finally, save all the metadata
+            mask_meta_df = mask_meta_df.sort_values(by=['file_name'])
+            mask_meta_df.to_csv(os.path.join(self.tile_dir, 'frames_meta.csv'),
+                                sep=',')
+            assert os.path.exists(os.path.join(self.tile_dir,
```
[the remark this answers is not in the document]
Comment (anithapk) [reply micro_dl/preprocessing/tile_images_uni_struct.py]: oops, that was a vestigial assert from debugging! sorry

**Pull request merged**